internal/pkg/template/diff/diff.go

```diff
@@ -90,7 +90,77 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	fromSeq, toSeq := make([]yaml.Node, len(from.Content)), make([]yaml.Node, len(to.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	cachedDiff, cachedErr := make(map[string]*Node), make(map[string]error)
```

Comment:
```suggestion
    type cacheEntry struct {
    	  node *Node
	      error error
    }
	cache := make(map[string]cacheEntry)
```
if the keys are the same and entries are always present in both?

Reply:
9Ehh since the size of `struct` is usually larger than the sum of the sizes of its fields, I tried to avoid doing this since there might potentially be a lot of diff. But after a quick look, it seems like in the case of pointer + error, the struct size isn't larger than the sum (guess it sort of depend on CPU arch though).